bulletroboy/roboy.py

```diff
@@ -0,0 +1,155 @@
+      closeEnough = False
+      iter = 0
+      dist2 = 1e30
+      while (not closeEnough and iter < maxIter):
+        jointPoses = p.calculateInverseKinematics(self.body_id, self.endEffectorId, targetPos)
+        #import pdb; pdb.set_trace()
+        # rclpy.logging._root_logger.info("resetting joints states")
+
+        for i in range(len(self.freeJoints)):
+          p.resetJointState(self.body_id, self.freeJoints[i], jointPoses[i])
+        ls = p.getLinkState(self.body_id, self.endEffectorId)
+        newPos = ls[4]
+        diff = [targetPos[0] - newPos[0], targetPos[1] - newPos[1], targetPos[2] - newPos[2]]
+        dist2 = (diff[0] * diff[0] + diff[1] * diff[1] + diff[2] * diff[2])
+        closeEnough = (dist2 < threshold)
+        iter = iter + 1
+      #print ("Num iter: "+str(iter) + "threshold: "+str(dist2))
+      return jointPoses
+
+    def drawDebugLines(self, targetPos):
+        # drawing debug lines
+        ls = p.getLinkState(self.body_id, self.endEffectorId)
+        if (self.hasPrevPose):
+            p.addUserDebugLine(self.prevPose, targetPos, [0, 0, 0.3], 1, self.trailDuration)
+            p.addUserDebugLine(self.prevPose1, ls[4], [1, 0, 0], 1, self.trailDuration)
+        self.prevPose = targetPos
+        self.prevPose1 = ls[4]
+        self.hasPrevPose = 1
+        
+    def joint_state_timer_callback(self):
+        """Callback function for the timer, publishes joint message every time it gets triggered by timer. 
+        """
+        msg = JointState()
+        msg.effort = [0.0]*len(self.joint_names)
+        msg.velocity = [0.0]*len(self.joint_names)
+        for i in range(p.getNumJoints(self.body_id)):
+            js = p.getJointState(self.body_id, i)
+            msg.position.append(js[0])
+            msg.name.append(self.joint_names[i])
+        self.joint_publisher.publish(msg)
+
+    def publish_collision(self, collision):
+        """Processes the collision message and publishes it.
+
+        Args:
+            collision: item of the contact points list output of pybullet getcontactPoints.
+        """
+        msg = Collision()
+
+        #collision[3] == linkIndexA in PyBullet docu
+        msg.linkid = collision[3]
+
+        #collision[5] == positionOnA in PyBullet docu
+        pos_in_lf = self.get_pos_in_link_frame(collision[3], collision[5])
+        msg.position.x = pos_in_lf[0]
+        msg.position.y = pos_in_lf[1]
+        msg.position.z = pos_in_lf[2]
+
+        #collision[7] == contactNormalOnB in PyBullet docu
+        msg.contactnormal.x = collision[7][0]
+        msg.contactnormal.y = collision[7][1]
+        msg.contactnormal.z = collision[7][2]
+
+        #collision[8] == contactDistance in PyBullet docu
+        msg.contactdistance = collision[8]
+
+        #collision[9] == normalForce in PyBullet docu
+        msg.normalforce = collision[9]
+
+        rclpy.logging._root_logger.info("Publishing collision in link %i" % msg.linkid)
```

Comment:
this will give you a global logger, but each node has it's own logger and when we can we should use that one:
```suggestion
        self.get_logger().info("Publishing collision in link %i" % msg.linkid)
```